examples/syscall-steal.c

```diff
@@ -43,6 +43,14 @@
 
 #if defined(CONFIG_KPROBES)
 #define HAVE_KPROBES 1
+#if defined(CONFIG_X86_64)
+ /* For x86 architecture, the syscall table cannot be used to invoke a syscall
+ * after commit 1e3ad78.
+ * Use a hook on the syscall entry instead to intercept the syscall.
+ * Alternatively, set USE_KPROBES_PRE_HANDLER_BEFORE_SYSCALL to 0 to use the old method.
+ */
+#define USE_KPROBES_PRE_HANDLER_BEFORE_SYSCALL 1
+#endif
 #include <linux/kprobes.h>
 #else
 #define HAVE_PARAM 1
```

Comment:
Also, we need to add the other condition, the system is v5.9+, and x86 arch without kprobe support.
Maybe we can tell the user they are unable to run this example.

Reply:
Maybe defining USE_KPROBES_PRE_HANDLER_BEFORE_SYSCALL to 0 by default is enough because users may turn on this macro if they get stuck when hacking.

Reply:
But the module will be broken if we have such a scenario.
Moreover, the user may be confused about this and start googling your question again.
I would suggest that you provide a more comprehensive explanation or relevant links (e.g., the stack overflow answer you mentioned).
Additionally, in the code, we can print out some warnings about this.